src/utils.ts

```diff
@@ -40,95 +32,86 @@ export const defaultRepo = (type) => {
   let api
   try {
     api = fs.readFileSync(path.join(repoPath, 'api'))
-  } catch (/** @type {any} */ err) {
+  } catch (err: any) {
     log('Unable to open api file')
   }
 
-  return api ? api.toString() : null
+  return (api != null) ? api.toString() : null
 }
 
-export const tmpDir = (type = '') => {
+export const tmpDir = (type = ''): string => {
   return path.join(os.tmpdir(), `${type}_ipfs_${nanoid()}`)
 }
 
-/**
- * @param {import('./types').ControllerOptions} opts
- */
-export function buildInitArgs (opts = {}) {
+export function buildInitArgs (opts: ControllerOptions = {}): string[] {
   const args = ['init']
-  const ipfsOptions = opts.ipfsOptions || {}
-  const initOptions = ipfsOptions.init && typeof ipfsOptions.init !== 'boolean' ? ipfsOptions.init : {}
+  const ipfsOptions: IPFSOptions = opts.ipfsOptions ?? {}
+  const initOptions = ipfsOptions.init != null && typeof ipfsOptions.init !== 'boolean' ? ipfsOptions.init : {}
 
   // default-config only for JS
   if (opts.type === 'js') {
-    if (ipfsOptions.config) {
+    if (ipfsOptions.config != null) {
       args.push(tempWrite.sync(JSON.stringify(ipfsOptions.config)))
     }
 
-    if (initOptions.pass) {
-      args.push('--pass', '"' + initOptions.pass + '"')
+    if (initOptions.pass != null) {
+      args.push('--pass', `"${initOptions.pass}"`)
     }
   }
 
   // Translate IPFS options to cli args
-  if (initOptions.bits) {
+  if (initOptions.bits != null) {
     args.push('--bits', `${initOptions.bits}`)
   }
 
-  if (initOptions.algorithm) {
+  if (initOptions.algorithm != null) {
     args.push('--algorithm', initOptions.algorithm)
   }
 
-  if (initOptions.emptyRepo) {
+  if (initOptions.emptyRepo === true) {
     args.push('--empty-repo')
   }
 
-  if (Array.isArray(initOptions.profiles) && initOptions.profiles.length) {
+  if (Array.isArray(initOptions.profiles) && initOptions.profiles.length > 0) {
     args.push('--profile', initOptions.profiles.join(','))
   }
 
   return args
 }
 
-/**
- * @param {import('./types').ControllerOptions} opts
- */
-export function buildStartArgs (opts = {}) {
-  const ipfsOptions = opts.ipfsOptions || {}
-  const customArgs = opts.args || []
+export function buildStartArgs (opts: ControllerOptions = {}): string[] {
+  const ipfsOptions: IPFSOptions = opts.ipfsOptions ?? {}
+  const customArgs: string[] = opts.args ?? []
 
   const args = ['daemon'].concat(customArgs)
 
   if (opts.type === 'js') {
-    if (ipfsOptions.pass) {
+    if (ipfsOptions.pass != null) {
       args.push('--pass', '"' + ipfsOptions.pass + '"')
     }
 
     if (ipfsOptions.preload != null) {
       args.push('--enable-preload', Boolean(typeof ipfsOptions.preload === 'boolean' ? ipfsOptions.preload : ipfsOptions.preload.enabled).toString())
     }
 
-    if (ipfsOptions.EXPERIMENTAL && ipfsOptions.EXPERIMENTAL.sharding) {
+    if ((ipfsOptions.EXPERIMENTAL != null) && ipfsOptions.EXPERIMENTAL.sharding === true) {
```

Comment:
`if (ipfsOptions.EXPERIMENTAL?.sharding === true) {`

Reply:
Good catch, this style of guarding is done by the `--fix` flag to the lint command.

```suggestion
    if (ipfsOptions.EXPERIMENTAL?.sharding === true) {
```